Parse/PFUser.m

```diff
@@ -652,23 +652,21 @@ - (NSDictionary *)RESTDictionaryWithObjectEncoder:(PFEncoder *)objectEncoder
                                             state:(PFObjectState *)state
                                 operationSetQueue:(NSArray *)queue
                                deletingEventually:(BOOL)isDeletingEventually {
-    @synchronized (self.lock) {
-        NSMutableArray *cleanQueue = [queue mutableCopy];
-        [queue enumerateObjectsUsingBlock:^(PFOperationSet *operationSet, NSUInteger idx, BOOL *stop) {
-            // Remove operations for `password` field, to not let it persist to LDS.
-            if (operationSet[PFUserPasswordRESTKey]) {
-                operationSet = [operationSet copy];
-                [operationSet removeObjectForKey:PFUserPasswordRESTKey];
-
-                cleanQueue[idx] = operationSet;
-            }
-        }];
-        return [super RESTDictionaryWithObjectEncoder:objectEncoder
-                                    operationSetUUIDs:operationSetUUIDs
-                                                state:state
-                                    operationSetQueue:cleanQueue
-                                   deletingEventually:isDeletingEventually];
-    }
+    NSMutableArray *cleanQueue = [queue mutableCopy];
+    [queue enumerateObjectsUsingBlock:^(PFOperationSet *operationSet, NSUInteger idx, BOOL *stop) {
+        // Remove operations for `password` field, to not let it persist to LDS.
+        if (operationSet[PFUserPasswordRESTKey]) {
+            operationSet = [operationSet copy];
```

Comment:
I feel like it's an implementation detail of the superclass method, so probably safer to leave it here.